py/shared/emails/main.py

```diff
@@ -264,26 +265,46 @@ async def send_event_update(self, action_id):
                 action_id,
                 self.settings.auth_key,
             )
+            group_messages = json.loads(data['group_messages'])
             user_tickets = await conn.fetch(
                 """
-                SELECT DISTINCT user_id, id AS ticket_id
+                SELECT DISTINCT user_id, id AS ticket_id, ticket_type
                 FROM tickets
                 WHERE status='booked' AND event=$1 AND user_id IS NOT NULL
                 """,
                 data['event_id'],
             )
 
-        ctx = {
-            'event_link': data['event_link'],
-            'event_name': data['event_name'],
-            'subject': data['subject'],
-            'message': data['message'],
-            'category_name': data['cat_name'],
-            is_cat(data['cat_slug']): True,
-        }
-        users = [UserEmail(id=user_id, ctx=ctx, ticket_id=ticket_id) for user_id, ticket_id in user_tickets]
+        messages = dict(
```

Comment:
If we really are going to go forward with a message per ticket type and a default message, please split this function from here on into two functions rather than combining all the logic into one here.

Reply:
This is about to be extended with new functionality. Will refactor when solution for updating donors is added.